src/Raven.Server/Documents/Indexes/IndexDefinitionBaseServerSide.cs

```diff
@@ -151,11 +151,11 @@ public static class IndexVersion
 
             public const long EngineTypeStored = 54_000; // introducing Corax, added engine type to the index storage
 
-            public const long GuaranteedOrderOfPropertiesInMapReduceIndexes = 54_001; // RavenDB-17312
+            public const long GuaranteedOrderOfGroupByFieldsInMapReduceIndexes = 54_002; // RavenDB-17312 - version 54_001 had an issue so we had to fix it and bump version again
```

Comment:
What will happen when the index is built on `54_001`?  Is it safe to switch them to the `legacy` version of the order?

Reply:
You, are right. I need to have the code using version `54_001` as well

Reply:
That is a bit tricky. In version `54_001` we ensured the same order of _all_ properties. Now we sort only group by fields. This means that the order of group by fields remained the same. So we can use the same code. But previously we sorted fields  _only_ if we had any map fields (in JS indexes we can have none):
https://github.com/ravendb/ravendb/blob/6a5cabb0d13fb576ca99d98ec9475519d36d59f9/src/Raven.Server/Documents/Indexes/MapReduce/Static/MapReduceIndex.cs#L490-L493

This means we still need to recognize this case. In https://github.com/ravendb/ravendb/pull/15822/commits/86fc1734e040def6a37a857d7e6fb3d01db4c7c9 I ensured the backward compatibility and added the test.